### ISSUE TITLE
EMFGeneratorFragment2: Improve fileHeader handling

### ISSUE THREAD

**Issue by miklossy** [eclipse/xtext-core#1345]
Currently, if the mwe2 workflow `fileHeader` configuration contains copyright headers starting with `/** `or `/***************************` and ending with` **/` or `***************************/` symbols, the copyright header of the generated EMF code looks a little bit weird.

See e.g the Domainmodel 
```
/**
 * *
 * Copyright (c) 2011, 2019 itemis AG (http://www.itemis.eu) and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 */
```
and the SimpleArithmetics
```
/**
 * ******************************************************************************
 * Copyright (c) 2015, 2019 itemis AG (http://www.itemis.eu) and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *  ******************************************************************************
 */
```
examples.

**Comment by cdietrich** [comment]
dont usual copyright headers look like

```
/*******************************************************************************
```

**Comment by cdietrich** [comment]
so i wonder is there is an option in emf

**Comment by miklossy** [comment]
CC @merks

**Comment by merks** [comment]
The template for all headers of all generated *.java files looks like this:

```
/**
<%{GenBase copyrightHolder = argument instanceof GenBase ? (GenBase)argument : argument instanceof Object[] && ((Object[])argument)[0] instanceof GenBase ? (GenBase)((Object[])argument)[0] : null;%>
<%if (copyrightHolder != null && copyrightHolder.hasCopyright()) {%>
 * <%=copyrightHolder.getCopyright(copyrightHolder.getGenModel().getIndentation(stringBuffer))%>
<%}}%>
 */
```

So the copyright in the model will always be inserted, line for line, with " * " prefixed before each line.

That's because the same copyright text is also inserted in *.properties files via this template:

```
<%{GenBase copyrightHolder = argument instanceof GenBase ? (GenBase)argument : argument instanceof Object[] && ((Object[])argument)[0] instanceof GenBase ? (GenBase)((Object[])argument)[0] : null;%>
<%if (copyrightHolder != null && copyrightHolder.hasCopyright()) {%># <%=copyrightHolder.getCopyright(copyrightHolder.getGenModel().getIndentation(stringBuffer))%>
<%} else {%>#
<%}}%>
```

Here each line is prefixed by "# ". 

And there is also this for plugin.xml:

```
<!--
<%{GenBase copyrightHolder = argument instanceof GenBase ? (GenBase)argument : argument instanceof Object[] && ((Object[])argument)[0] instanceof GenBase ? (GenBase)((Object[])argument)[0] : null;%>
<%if (copyrightHolder != null && copyrightHolder.hasCopyright()) {%>
 <%=copyrightHolder.getCopyright(copyrightHolder.getGenModel().getIndentation(stringBuffer))%>
<%}}%>
-->
```

**Comment by cdietrich** [comment]
@merks so there is no option for a copyright header as the foundation proposes

**Comment by merks** [comment]
That the answer is no is clear from the template.  Of course dynamic templates are an option, though not a great option.

**Comment by cdietrich** [comment]
Yes introducing dynamic templates by default is a terrible idea so there is no viable option

**Comment by cdietrich** [comment]
As emf does not have multi star headers and it seems ok I am ok with proceeding

**Comment by miklossy** [comment]
Fixed in 2.21.
